### ISSUE TITLE
System freeze - NVIDIA

### ISSUE THREAD

**Issue by Yuri6037** [pop-os/pop#1750]
**Distribution (run `cat /etc/os-release`):**
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
NVIDIA 465.31

**Issue/Bug Description:**
I have finally figured out what exactly is the random crash on my desktop workstation which uses a GTX 650 Ti. It turns out in order to get proper logs you must use `sudo journalctl -b -1 -r`. Without sudo you won't get any information from NVIDIA driver.

So the crash is in fact NVIDIA related and is a Xid 79 error; the famous "GPU has fallen off the bus.". And guess what: Oryx Pro is also affected by Xid 79 however the error it produces includes something weird about power state. So perhaps Oryx is a different bug but at least now we know the reason why GTX 650 is crashing.

Logs from Oryx Pro which includes the weird additional log before reaching Xid 79 like my workstation:
```
Jun 28 15:23:30 oryp7 kernel: NVRM: GPU 0000:01:00.0: GPU has fallen off the bus.
Jun 28 15:23:30 oryp7 kernel: NVRM: Xid (PCI:0000:01:00): 79, pid=2337, GPU has fallen off the bus.
Jun 28 15:23:30 oryp7 kernel: NVRM: GPU at PCI:0000:01:00: GPU-35195e53-762e-fc41-e9b5-123a60283c15
Jun 28 15:23:30 oryp7 kernel: nvidia 0000:01:00.0: can't change power state from D3cold to D0 (config space inaccessible)
```

after this you get an infinite loop of "kernel: nvidia-modeset: ERROR: GPU:0: Error while waiting for GPU progress: 0x0000c67d:0 2:0:4048:4040".

On my workstation only the like which says Xid 79 GPU fallen off the bus appears.

According to NVIDIA Xid 79 could be temps, power, gpu or PCI hardware defect or even BIOS/firmware issue.

There are forums which also mentions a bug with some ASUS motherboards. Unfortunatly the workstation is running an ASUS board.

**Steps to reproduce (if you know):**
No particular steps produces this, it's absolutely random. May occur after resume from suspend, may occur after locking GNOME session, may occur after a load on the GPU, may even occur while running a high load on the GPU...

**Expected behavior:**
No crash.

**Other Notes:**
This is a follow up of my older issue which is still unsolved.

**Comment by bndF1** [comment]
I'm also having this issue.

I've opened a Reddit thread with some journal logs.

https://www.reddit.com/r/pop_os/comments/o8dlh8/random_freezes_help/?utm_medium=android_app&utm_source=share

**Comment by Yuri6037** [comment]
I'm not sure if your issue is the same as this one. I don't see any power state issue in the logs on that reddit. I'm gonna do some tests to attempt to reproduce the crash.

**Comment by bndF1** [comment]
Wow, maybe I misunderstood, sorry.
My problema is that doing whatever, computer might get frozen.
And it's very annoying because I need stability to work.
Reboot doesn't mean, that the system is not going to freeze, it could happen whenever.

Thanks

**Comment by Yuri6037** [comment]
The nvidia power state failure is actually already reported over here: https://github.com/system76/firmware-open/issues/160. I don't think it's random.

After a lot more analysis, it appears D3cold is problematic for many PCIe bridge/device combo. There are also people who get D3cold failure with ethernet cards! However research seem to indicate D3cold never fails on desktop computers it only fails on laptops especially laptops with dual GPU config (Intel + NVIDIA).

And there's also a NVIDIA dev forum about this issue: https://forums.developer.nvidia.com/t/bug-cant-change-power-state-from-d3cold-to-d0-config-space-inaccessible-stuck-at-boot/112912/4. For NVIDIA it is a system firmware bug.

For now the crash does not seem to trigger as long as there's something running on the dGPU. From what I can see as long as something is using the dGPU, there shouldn't be any crash. Booting in Compute mode does not start this Xorg service in nvidia-smi causing the crash. I suspect Xorg randomly crashes when you lock the GNOME session or even better GNOME session closes the Xorg instance on the dGPU causing the crash. When you sleep/resume I suspect Xorg also goes down which basically gives a chance to nvidia to crash with the power state failure.
I actually compiled and ran gpu-burn with both doubles and tensor cores (AC plugged in + high performance mode of course) on my RTX 3060 in Hybrid Graphics and it never crashed, the fans started to spin like hell but no crash; temp got as high as 80C not more, which is still in spec for both Intel and NVIDIA so it can't be a temp or power issue like NVIDIA claims it is for Xid 79.
I kept an eye on dGPU status with nvidia-smi + constantly leaving "NVIDIA X Server Settings" open to make sure nothing was going too bad.

**Comment by Yuri6037** [comment]
The issue has just occured again on 21.04. It's just much less frequent. The error log shows the exact same error reguarding NVIDIA power states and XID 79. However now I have enough data to say that the crash always occurs after pressing the windows key + L to lock GNOME session and having rhythmbox running in the background with a bluetooth headset on and connected. What's strange is the crash never occured while using the GPU the crash wlays occured at the exact moment you type Win + L keys. Even more weird, I couldn't see the crash happening without my bluetooth headset.

I think I found something: NVIDIA takes some time to wake from screen lock same as when the computer was sleeping. Rhythmbox is sending notifications everytime the track changes or the play/pause button is triggered. GNOME is waking the screen everytime it sees a new notification.
Is there any chance that NVIDIA does not like GNOME waking up the screen immediatly after it was shut down?
This would explain the randomness of the crash. You need to lock the screen at the EXACT moment Rhythmbox wants to show a notification otherwise the crash would not occur.

**Comment by JoonasC** [comment]
@Yuri6037 What nvidia driver version are you running? What are the CUDA NVIDIA X Settings?

**Comment by Yuri6037** [comment]
The problem happened again, this time I've just found a new method to trigger it:

- Start laptop in hybrid mode,

- Connect a bluetooth mouse (such as MS Precision Mouse or Apple Magic Mouse)

- Connect external HDMI 4K display,

- Configure computer to output 4K 60Hz and treat external display as extended,

- Open audio configuration on laptop and set it to output to NVIDIA HDMI, play some sound to make sure the TV is actually outputting sound from laptop,

- Apply and save settings, you should now have the TV showing a duplicate of your desktop,

- Close laptop lid,

- The TV screen will now bug as hell for maybe a few seconds, wait for it to stabilize,

- After it stabilized mouse cursor will start moving smoothly again,

- Re-open laptop lid,

- Wait for both screens stabilize again (GNOME/Xorg moving windows from one screen to the other),

- Disconnect HDMI cable (leaving all configuration as is), at this point the laptop should still work,

- Immediatly close laptop lid,

- 2 seconds later enjoy fan spinning as hell and dead black screen (NVIDIA has died again),

- Force reboot computer (use the power button),

- Once in gnome open a terminal and type `sudo journalctl -r -b -1`, read the log and enjoy NVIDIA Xid79 before NVIDIA "can't change power state from d3cold to d0 (config space inaccessible)"

**Comment by cstrahan** [comment]
Also running into this exact problem. The laptop (gaze16) crashed about 10 times or so while working through these comments (yes, I power cycled that many times), until I gave up and used a different computer to continue reading and finally write this comment. Been trying to resolve this for the past 4 hours, and my hopes for squeezing in some software development on my side project are just about dashed.

**Comment by cstrahan** [comment]
Running on AC, the laptop seems to be very stable.

Running on battery ("balanced" profile, fwiw), the laptop freezes very quickly after boot. Some particulars:
- If I power cycle, enter my LUKS password, and walk away for a couple seconds (while the GDM login screen is up), it pretty much 100% freezes within about 5 seconds.
- If I instead log in, this prolongs how long it takes to freeze. From there, some additional actions impact how much time I have before it freezes. For example, I experimented with playing a you YouTube video in Firefox, and that seemed to give me maybe 20-30 seconds before freezing. If instead the first and only thing I do is fire up gnome-terminal and view `journalctl`, I maybe get 15 seconds or so.

My gaze16 is effectively a very expensive and under powered laptop at the moment :cry:

**Comment by cstrahan** [comment]
If I unplug power to the laptop, it hangs in a matter of seconds. Interestingly, if I reconnect power the fans spin up to (what seems to be) 100%, and remains hung. Only "fix" seems to be power cycling.

This is on the gaze16, Pop!_OS 21.10, with latest nvidia drivers (~470, IIRC) and all packages updated, with latest firmware.

**Comment by leviport** [comment]
@cstrahan have you opened a support ticket already? Since you have System76 hardware, the support team is available to help you with this.

Also, is the machine still in the same configuration that it was in when you received it, or have you changed any components? Have you already tested your RAM to make sure that's not the issue?

**Comment by cstrahan** [comment]
@leviport 

> have you opened a support ticket already?

No. I was thinking about doing so, but ultimately, I decided on doing a clean reinstall of the OS. I had troubles upgrading from 21.10 to 22.04 (the problem described here: https://github.com/pop-os/pop/issues/2410), so though I managed to continue with the upgrade, I suspected that some things were still not as they should be.

Things seem fine now after the clean install, AFAICT.

> Also, is the machine still in the same configuration that it was in when you received it, or have you changed any components? Have you already tested your RAM to make sure that's not the issue?

I did replace RAM with larger modules: [TEAMGROUP Elite DDR4 64GB Kit (2 x 32GB) 3200MHz PC4-25600 CL22 Unbuffered Non-ECC 1.2V SODIMM 260-Pin Laptop Notebook PC Computer Memory Module Ram Upgrade - TED464G3200C22DC-S01 ](https://www.amazon.com/gp/product/B08TQBY2NR/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)

I ran [Memtest86+](https://www.memtest.org/) and didn't see any problems. Only had issues when on battery and running with nvidia graphics.

**Comment by leviport** [comment]
Glad to hear that a reinstall fixed it. That pretty well rules out hardware issues.

**Comment by dsilas** [comment]
I am also getting Xid 79 issues in pop 20.04 on my galp5 and have been on and off for some months in hybrid and nvidia mode, sometimes on resuming, sometimes in the middle of working. It seems to happen both on AC and battery. I am in contact with support. @cstrahan did your issues ever reemerge after the clean install?

**Comment by cstrahan** [comment]
@dsilas My system has been pretty stable since the last clean install :+1:

**Comment by valiantgenomics** [comment]
@dsilas Just received this error `nvidia-modeset: ERROR: GPU:0 failed detecting connected display devices` similar to you after switching my graphics to hybrid in BIOS. I received that error when my battery died and the laptop tried shutting down.

I changed them to hybrid because the cryptdata screen was flashing multiple times on boot but by changing to hybrid graphics in BIOS and leaving PopOS to NVIDIA graphics resolved that flashing issue.

Do you think that `nvidia-modeset: ERROR: GPU:0 failed detecting connected display devices` is caused by the hybrid graphics setting?

Thanks! I'm new to PopOS and it's the best distro I've had the pleasure to run! It's my daily driver!

**Comment by majal** [comment]
I have gaze16 and have the same problem on Ubuntu 22.04. Does not wake up on suspend. Happens consistently if the following two conditions are true:

 - GPU on Hybrid mode
 - There is a running application that is using the GPU when suspend happened.

From the reports, it seems that this is not a System76-specific problem. But it would be nice if S76 can bug NVIDIA about this.

**Comment by armangurkan** [comment]
same here, the hardware was changed 1 time already

**Comment by logicbomb421** [comment]
Another user reporting in. Actually happened while I was trying to write this reply.

Running PopOS 22.04 using kernel 6.0.12 on a gaze16-3060 with System76 driver version 20.04.72 and NVIDIA driver 525.

Going to open an actual support case with S76. If they are able to resolve it, I'll post another comment here with info.

**Comment by bym0** [comment]
If someone is interested:
In many reddit threads and gh issues, i read that they were suspecting gnome-shell to be the culprit.

I can sort of proof that, today i switched/tested KDE (just installed `kde-plasma-desktop` and rebooted) and voila - i have no crashes/freezes.

**Comment by dsilas** [comment]
@bym0 yes I was wondering about that too based on a few log messages, and I'm getting similar result to you. I tried xfce from the  package manager in pop 20.04 
`sudo apt install xfce4 xfce4-goodies`
And kept gdm as the display manager.
(anyone looking for directions see here: https://support.system76.com/articles/desktop-environment/) 
I had the graphics set to compute since that's usually how I use the card:

- If I login with the Xfce session, I can suspend and wake the computer
- If I login to the default pop/gnome session, I can suspend but I _cannot_ wake up.

Maybe next time i have a moment  I'll see if disabling all the gnome extensions helps.
